chapters/06_agent_based_extension/01_state_dependent/main.tex

```diff
@@ -0,0 +1,148 @@
+\section{State-dependent variation}\label{sec:state_dependent}
+
+This section focuses on creating a state-dependent variation of the queueing
+system described in Section~\ref{sec:queueing_section}.
+The state-dependent version is based on the idea that the speed of the servers
+could be based on the number of individuals present in the system.
+This was implemented only on the Discrete Event Simulation (DES) version of
+the queueing model described in Section~\ref{sec:discrete_event_simulation}.
+
+The new service rate \(\mu\) that is used here is defined as follows:
+
+\begin{equation}\label{eq:state_dependent_service_rate}
+    \mu =
+    \begin{cases}
+        \mu_{(0,0)}, & \text{if } (u, v) = (0, 0) \\
+        \mu_{(0,1)}, & \text{if } (u, v) = (0, 1) \\
+        \quad \vdots & \qquad \quad \vdots \\
+        \mu_{(N,M)}, & \text{if } (u, v) = (M, N)
+    \end{cases}
+\end{equation}
+
+Although the state-dependent variation of the queueing system was implemented
+only on the DES version of the queue, for visualisation purposes, the Markov
+chain model will be used to better demonstrate the concept of this model.
```

Comment:
I think you just don't need to mention this yet. The model is conceptual at the moment.

Reply:
I wasn't planning on saying that, but I do use the Markov model figure somewhere and kinda felt that I should mention something that the Markov model is not actually part of the agent-based model and was a bit out of the blue.

This is the picture that I use 👇 which I think is helpful with visualisation:

![image](https://user-images.githubusercontent.com/20097997/218537037-43f0ce53-b0c9-4c99-a734-fadb23f22a47.png)

Reply:
> I wasn't planning on saying that, but I do use the Markov model figure somewhere and kinda felt that I should mention something that the Markov model is not actually part of the agent-based model and was a bit out of the blue.


The Markov model is the basis for the abs model. It's just that you don't get the steady state probabilities analytically at any point. The conceptual model that this is a picture of is still there.